Analysis/Tasks/PWGGA/myGammaTask3.cxx

```diff
@@ -0,0 +1,281 @@
+        {
+            if( theTrack.tpcNSigmaEl() > fPIDnSigmaBelowElectronLine && theTrack.tpcNSigmaEl() < fPIDnSigmaAboveElectronLine && theTrack.tpcNSigmaPi()<fPIDnSigmaAbovePionLine)
+            {
+                registry.fill(HIST("IsPhotonSelected"), kPionRejLowMom);
+                return kFALSE;
+            }
+        }
+        // High Pt Pion rej
+        else
+        {
+            if( theTrack.tpcNSigmaEl() > fPIDnSigmaBelowElectronLine && theTrack.tpcNSigmaEl() < fPIDnSigmaAboveElectronLine && theTrack.tpcNSigmaPi()<fPIDnSigmaAbovePionLineHighPt)
+            {
+                registry.fill(HIST("IsPhotonSelected"), kPionRejHighMom);
+                return kFALSE;
+            }
+        }
+    }
+    return kTRUE;  
+}
+
+
+
+void ATask::init(InitContext const&)
+{
+    TAxis *lXaxis = registry.get<TH1>(HIST("IsPhotonSelected"))->GetXaxis();
+    for (size_t i=0; i< fPhotCutsLabels.size(); ++i)
+    {
+        lXaxis->SetBinLabel(i+1, fPhotCutsLabels[i]);
+    }
+}
+  
+void ATask::process(aod::Collision const& collision, 
+                    aod::V0Datas const& v0s, 
+                    tracksAndTPCInfo const& tracks,
+                    aod::McParticles const& mcParticles)
+{
+    for (auto& v0 : v0s) {
+        ++i;
+        float lPhiV0Rec = static_cast<float>(M_PI) + std::atan2(-v0.py(), -v0.px());
+        
+        // fill some QA histograms before any cuts
+        registry.fill(HIST("beforeCuts/hPtRec_before"), v0.pt());
+        registry.fill(HIST("beforeCuts/hEtaRec_before"), v0.eta());
+        registry.fill(HIST("beforeCuts/hPhiRec_before"), lPhiV0Rec);
+        registry.fill(HIST("beforeCuts/hConvPointR_before"), v0.v0radius());
+
+        registry.fill(HIST("IsPhotonSelected"), kPhotonIn);
+        
+        auto trackPos = v0.template posTrack_as<tracksAndTPCInfo>(); //positive daughter
+        auto trackNeg = v0.template negTrack_as<tracksAndTPCInfo>(); //positive daughter
+        
+        // single track eta cut
+        if (TMath::Abs(trackPos.eta()) > fEtaCut || TMath::Abs(trackNeg.eta()) > fEtaCut)
+        {
+            registry.fill(HIST("IsPhotonSelected"), kTrackEta);
+            continue;
+        }
+        
+        // single track pt cut    
+        if ( (TMath::Power(v0.pxpos(),2) + TMath::Power(v0.pypos(),2) < fSinglePtSq) ||
+             (TMath::Power(v0.pxneg(),2) + TMath::Power(v0.pyneg(),2) < fSinglePtSq)
+           )
+        {
+            registry.fill(HIST("IsPhotonSelected"), kTrackPt);
+            continue;
+        } 
+        
+        if (!(selectionPIDTPC_track(trackPos) && selectionPIDTPC_track(trackNeg)))
+        {
+            continue;
+        }
+        
+        if (v0.v0radius() < fMinR || v0.v0radius() > fMaxR)
+        {
+                registry.fill(HIST("IsPhotonSelected"), kV0Radius);
+                continue;
+        }
+        
+        Double_t lAlpha = -1.;
+        Double_t lQt = -1.;
+        computeArmenteros(v0, lAlpha, lQt);
```

Comment:
The Armenteros-Podolanski quantities are already computed and available in the V0Datas table. Please use those. You can access them via: v0.alpha(), v0.qtarm().